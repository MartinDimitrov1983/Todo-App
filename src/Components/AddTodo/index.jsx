import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '..';
import { ADD_TODO } from '../../Constants';
import styles from './index.module.css';

const AddTodo = ({ onClick, onInputChange, value }) => {
  return (
    <div className={styles.container}>
      <input
        className={styles.input}
        placeholder={ADD_TODO}
        value={value}
        onChange={onInputChange}
      />
      <Button text={ADD_TODO} onClick={onClick}></Button>
    </div>
  );
};

AddTodo.propTypes = {
  onClick: PropTypes.func,
  onInputChange: PropTypes.func,
  value: PropTypes.string,
};

export default AddTodo;
