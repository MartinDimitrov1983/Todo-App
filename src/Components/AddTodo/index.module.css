.container {
  display: flex;
  flex-direction: column;
  margin-top: 20px;
  gap: 10px;
}

.input {
  min-height: 30px;
  border-radius: 5px;
  padding: 0 10px;
}
