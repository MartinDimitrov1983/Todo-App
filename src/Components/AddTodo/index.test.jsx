import React from 'react';
import { render } from '@testing-library/react';

import AddTodo from './index';

test('Snapshot AddTodo', () => {
  const { asFragment } = render(
    <AddTodo onClick={() => {}} onInputChange={() => {}} value="Test" />,
  );
  expect(asFragment()).toMatchSnapshot();
});
