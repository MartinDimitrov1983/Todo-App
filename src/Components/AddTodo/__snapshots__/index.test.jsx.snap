// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot AddTodo 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <input
      class="input"
      placeholder="Add Todo"
      value="Test"
    />
    <button
      class="button"
      type="button"
    >
      <div>
        Add Todo
      </div>
    </button>
  </div>
</DocumentFragment>
`;
